python/lsst/pipe/base/config.py

```diff
@@ -209,6 +214,12 @@ class _BaseDatasetTypeConfig(pexConfig.Config):
                            doc="name of the DatasetType")
     storageClass = pexConfig.Field(dtype=str,
                                    doc="name of the StorageClass")
+    nameTemplate = pexConfig.Field(dtype=str,
+                                   default='',
+                                   optional=True,
+                                   doc=("Templated name of string, used to set name "
+                                        "field according to a shared substring at "
+                                        "configuration time"))
```

Comment:
Maybe change "configuration time" to "when `formatTemplateNames` is called", to be a bit clearer.